File: shoebill_plugins/AdminPlugin/src/main/java/lt/ltrp/dialog/ServerWeaponListDialog.java
package lt.ltrp.dialog;

import lt.ltrp.ItemPlugin;
import lt.ltrp.PlayerPlugin;
import lt.ltrp.data.LtrpWeaponData;
import lt.ltrp.object.*;
import net.gtaun.shoebill.Shoebill;
import net.gtaun.shoebill.common.dialog.ListDialog;
import net.gtaun.shoebill.constant.WeaponModel;
import net.gtaun.shoebill.resource.ResourceManager;
import net.gtaun.util.event.EventManager;

/**
 * @author Bebras
 *         2016.05.30.
 */
public class ServerWeaponListDialog {

    public static ListDialog create(LtrpPlayer player, EventManager eventManager) {
        return create(player, eventManager, null);
    }

    public static ListDialog create(LtrpPlayer player, EventManager eventManager, WeaponModel model) {
        PlayerPlugin playerPlugin = ResourceManager.get().getPlugin(PlayerPlugin.class);
        ItemPlugin itemPlugin = ResourceManager.get().getPlugin(ItemPlugin.class);
        return ListDialog.create(player, eventManager)
                .caption("Serverio ginklø perþiûra.")
                .item("Automobiliuose esantys ginklai", i -> {
                    new Thread(() -> {
                        WeaponItem[] items;
                        if(model == null)
                            items = itemPlugin.getItemDao().getWeaponItems(LtrpVehicle.class);
                        else
                            items = itemPlugin.getItemDao().getWeaponItems(model, LtrpVehicle.class);
                        LtrpWeaponData weaponData[] = new LtrpWeaponData[items.length];
                        int c = 0;
                        for(WeaponItem item : items)
                            weaponData[c++] = item.getWeaponData();
                        showWeaponsOnSampThread(player, eventManager, weaponData);
                    }).start();
                })
                .item("Namuose/garaþuose esantys ginklai", i -> {
                    new Thread(() -> {
                        WeaponItem[] hitems;
                        WeaponItem[] gitems;
                        if(model == null) {
                            hitems = itemPlugin.getItemDao().getWeaponItems(House.class);
                            gitems = itemPlugin.getItemDao().getWeaponItems(Garage.class);
                        } else {
                            hitems = itemPlugin.getItemDao().getWeaponItems(model, House.class);
                            gitems = itemPlugin.getItemDao().getWeaponItems(model, Garage.class);
                        }
                        LtrpWeaponData weaponData[] = new LtrpWeaponData[hitems.length + gitems.length];
                        int c = 0;
                        for(WeaponItem item : gitems)
                            weaponData[c++] = item.getWeaponData();
                        for(WeaponItem item : hitems)
                            weaponData[c++] = item.getWeaponData();
                        showWeaponsOnSampThread(player, eventManager, weaponData);
                    }).start();
                })
                .item("Þaidëjø kuprinëse esantys ginklai", i -> {
                    new Thread(() -> {
                        WeaponItem[] items;
                        if(model == null)
                            items = itemPlugin.getItemDao().getWeaponItems(LtrpPlayer.class);
                        else
                            items = itemPlugin.getItemDao().getWeaponItems(model, LtrpPlayer.class);
                        LtrpWeaponData weaponData[] = new LtrpWeaponData[items.length];
                        int c = 0;
                        for(WeaponItem item : items)
                            weaponData[c++] = item.getWeaponData();
                        showWeaponsOnSampThread(player, eventManager, weaponData);
                    }).start();
                })
                .item("Þaidëjø laikomi ginklai(darbiniai neskaièiuojami)", i -> {
                    new Thread(() -> {
                        showWeaponsOnSampThread(player, eventManager, playerPlugin.getPlayerWeaponDao().get());
                    }).start();
                })
                .buttonOk("Perþiûrëti")
                .buttonCancel("Uþdaryti.")
                .onClickOk((d, i) -> {
                    player.sendMessage("Informacijos surinkimas gali uþtrukti. Jûs uþðaldytas, gavus duomenis galësite vël judëti.");
                    player.toggleControllable(false);
                })
                .build();
    }

    private static void showWeaponsOnSampThread(LtrpPlayer player, EventManager eventManager, LtrpWeaponData[] weaponData) {
        Shoebill.get().runOnSampThread(() -> {
            new WeaponListDialog(player, eventManager, weaponData)
                    .show();
            player.toggleControllable(true);
        });
    }

}

File: shoebill_plugins/AdminPlugin/src/main/java/lt/ltrp/dialog/ServerWeaponListDialog.kt
package lt.ltrp.dialog;

import lt.ltrp.ItemPlugin;
import lt.ltrp.PlayerPlugin;
import lt.ltrp.`object`.Garage
import lt.ltrp.data.LtrpWeaponData;
import lt.ltrp.`object`.LtrpPlayer
import lt.ltrp.`object`.LtrpVehicle
import lt.ltrp.`object`.WeaponItem
import lt.ltrp.house.`object`.House
import lt.maze.dialog.ListDialog
import lt.maze.dialog.ListDialogItem
import net.gtaun.shoebill.Shoebill;
import net.gtaun.shoebill.constant.WeaponModel;
import net.gtaun.shoebill.resource.ResourceManager;
import net.gtaun.util.event.EventManager;

/**
 * @author Bebras
 *         2016.05.30.
 */

object ServerWeaponListDialog {

    fun create(player: LtrpPlayer, eventManager: EventManager, model: WeaponModel?): lt.maze.dialog.ListDialog {
        return ListDialog.create(player, eventManager, {
            val playerPlugin = ResourceManager.get().getPlugin(PlayerPlugin::class.java)
            val itemPlugin = ResourceManager.get().getPlugin(ItemPlugin::class.java)
            caption { "Serverio ginklø perþiûra." }
            item(ListDialogItem.create {
                itemText { "Automobiliuose esantys ginklai" }
                selectHandler {
                    Thread( {
                        val items : Array<WeaponItem>
                        if(model == null)
                            items = itemPlugin.getItemDao().getWeaponItems(LtrpVehicle::class.java)
                        else
                            items = itemPlugin.getItemDao().getWeaponItems(model, LtrpVehicle::class.java)

                        val weapons = items.map { it.weaponData }
                        showWeaponsOnSampThread(player, eventManager, weapons)
                    }).start()
                }
            })
            item(ListDialogItem.create {
                caption { "Namuose/garaþuose esantys ginklai" }
                selectHandler {
                    Thread({
                        val hitems: Array<WeaponItem>
                        val gitems: Array<WeaponItem>
                        if(model == null) {
                            hitems = itemPlugin.getItemDao().getWeaponItems(House::class.java)
                            gitems = itemPlugin.getItemDao().getWeaponItems(Garage::class.java)
                        } else {
                            hitems = itemPlugin.getItemDao().getWeaponItems(model, House::class.java)
                            gitems = itemPlugin.getItemDao().getWeaponItems(model, Garage::class.java)
                        }
                        showWeaponsOnSampThread(player, eventManager, hitems.plus(gitems).map { it.weaponData });
                    }).start();
                }
            })
            item(ListDialogItem.create {
                caption { "Þaidëjø kuprinëse esantys ginklai" }
                selectHandler {
                    Thread({
                        val items: Array<WeaponItem>
                        if(model == null)
                            items = itemPlugin.getItemDao().getWeaponItems(LtrpPlayer::class.java)
                        else
                            items = itemPlugin.getItemDao().getWeaponItems(model, LtrpPlayer::class.java)
                        showWeaponsOnSampThread(player, eventManager, items.map { it.weaponData })
                    }).start()
                }
            })
            item(ListDialogItem.create {
                caption { "Þaidëjø laikomi ginklai(darbiniai neskaièiuojami)" }
                selectHandler {
                    Thread( {
                        showWeaponsOnSampThread(player, eventManager, playerPlugin.getWeaponDao().get().toList())
                    }).start()
                }
            })
            onSelectItem { dialog, item ->
                player.sendMessage("Informacijos surinkimas gali uþtrukti. Jûs uþðaldytas, gavus duomenis galësite vël judëti.")
                player.toggleControllable(false)
            }
            buttonCancel { "Uþdaryti" }
            buttonOk { "Perþiûrëti" }
        })
    }

    fun create(player: LtrpPlayer, eventManager: EventManager): ListDialog {
        return create(player, eventManager, null)
    }

    private fun showWeaponsOnSampThread(player: LtrpPlayer, eventManager: EventManager, weapons: List<LtrpWeaponData>) {
        Shoebill.get().runOnSampThread {
            WeaponListDialog(player, eventManager, weapons.toTypedArray())
                .show()
            player.toggleControllable(true)
        }
    }
}
